biz.aQute.bndlib/src/aQute/bnd/osgi/repository/XMLResourceParser.java

```diff
@@ -0,0 +1,253 @@
+package aQute.bnd.osgi.repository;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+import org.osgi.resource.Resource;
+
+import aQute.bnd.header.Attrs;
+import aQute.bnd.osgi.Processor;
+import aQute.bnd.osgi.resource.CapReqBuilder;
+import aQute.bnd.osgi.resource.ResourceBuilder;
+
+public class XMLResourceParser extends Processor {
```

Comment:
No, I think it can grow to be a nice parser to be used in the Indexed Repositories. I also hope one day to let bnd replace repoindex. This is just a very handy utility